src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -88,7 +88,7 @@ object Parsers {
     def syntaxError(msg: String, offset: Int = in.offset): Unit =
       if (offset > lastErrorOffset) {
         syntaxError(msg, Position(offset))
-        lastErrorOffset = in.offset
+        lastErrorOffset = offset
```

Comment:
No, I think here you need to leave in.offset. (also in JavaParsers). lastErrorOffset is the offset we skipped to (as its doc comment implies). That change explains the test failures we are seeing.